src/context/userNotification.js

```diff
@@ -0,0 +1,40 @@
+import React, {createContext, useContext, useCallback, useMemo, useEffect} from "react"
+import PropTypes from "prop-types"
+
+import {UserContext} from "./user"
+import {NotificationContext} from "./notification"
+
+const UserNotificationContext = createContext()
+
+const UserNotificationProvider = ({children}) => {
+    const {notification, setNotification} = useContext(NotificationContext)
+    const {user: {firstName} = {}} = useContext(UserContext)
+
+    const prefix = useMemo(() => {
+        return firstName ? `${firstName}: ` : ""
```

Comment:
This will have the same issue we discussed in the slack, right? `prefix` will be empty until `firstName` changes, but then after you click log out the `firstName` is empty again so this will reset to `""`?

Reply:
Yeah it will after the logout since it is cached in useCallback. So while showing it will work but if you try after a render after logout, it will not show. But that totally depends on the use case. In this case we are showing a notification immediately. We are not waiting for a re-render. Like you said, we can call show notification in a useEffect instead of making the call directly.